lib/src/lib/utils/utils.ts

```diff
@@ -10,3 +11,86 @@ export function readPropValue<T, R = string>(value: propValue<T, R>, data?: T):
 export function coerceObservable<T = unknown>(value: T | Observable<T>): Observable<T> {
   return isObservable(value) ? value : of(value);
 }
+
+export type ReactiveOption<PAYLOAD, RETURN_TYPE> =
+  | RETURN_TYPE
+  | ((data?: PAYLOAD) => RETURN_TYPE | Signal<RETURN_TYPE>)
+  | Signal<RETURN_TYPE>;
+
+export type ReactiveBooleanOption<PAYLOAD> = ReactiveOption<PAYLOAD, boolean>;
+export type ReactiveStringOption<PAYLOAD> = ReactiveOption<PAYLOAD, string>;
+export type ReactiveNumberOption<PAYLOAD> = ReactiveOption<PAYLOAD, number>;
+
+export function computeReactiveBooleanOption<PAYLOAD>(
+  option: ReactiveBooleanOption<PAYLOAD> | undefined,
+  value: Signal<any>
+): boolean {
+  let response = false;
+  let optionCopy = option;
```

Comment:
should this not be `  let optionCopy = {...option};`?
because now it will refer to the exact same object reference. there is no difference between optionCopy and option

Reply:
same applies to other compute methods